File: kotlin/eliuds-eggs/src/main/kotlin/EliudsEggs.kt
object EliudsEggs {

    fun eggCount(number: Int): Int{
        var count = 0
        var num = number
        while (num != 0) {
            if (num and 1 > 0) count++
            num = num shr 1
        }
        return count
    }
}

File: fortran/eliuds-eggs/1/eliuds_eggs.f90
module eliuds_eggs

  implicit none

contains

  integer function eggCount(number)
    integer :: number, remaining

    eggCount = 0
    remaining = number
    do while (remaining > 0)
      eggCount = eggCount + iand(remaining, 1)
      remaining = shiftr(remaining, 1)
    end do
  end function

end module
